frontend/test/metabase/scenarios/models/model-actions.cy.spec.js

```diff
@@ -0,0 +1,201 @@
+import {
+  enableActionsForDB,
+  modal,
+  popover,
+  restore,
+} from "__support__/e2e/helpers";
+import { SAMPLE_DB_TABLES } from "__support__/e2e/cypress_data";
+import { createMockActionParameter } from "metabase-types/api/mocks";
+
+const PG_DB_ID = 2;
+
+const SAMPLE_ORDERS_MODEL = {
+  name: "Order",
+  dataset: true,
+  display: "table",
+  query: {
+    "source-table": SAMPLE_DB_TABLES.STATIC_ORDERS_ID,
+  },
+};
+
+const TEST_PARAMETER = createMockActionParameter({
+  id: "49596bcb-62bb-49d6-a92d-bf5dbfddf43b",
+  name: "Total",
+  slug: "total",
+  type: "number/=",
+  target: ["variable", ["template-tag", "total"]],
+});
+
+const TEST_TEMPLATE_TAG = {
+  id: TEST_PARAMETER.id,
+  type: "number",
+  name: TEST_PARAMETER.slug,
+  "display-name": "Total",
+  slug: TEST_PARAMETER.slug,
+};
+
+const SAMPLE_QUERY_ACTION = {
+  name: "Demo Action",
+  type: "query",
+  parameters: [TEST_PARAMETER],
+  database_id: PG_DB_ID,
+  dataset_query: {
+    type: "native",
+    native: {
+      query: `DELETE FROM orders WHERE total < {{ ${TEST_TEMPLATE_TAG.name} }}`,
+      "template-tags": {
+        [TEST_TEMPLATE_TAG.name]: TEST_TEMPLATE_TAG,
+      },
+    },
+    database: PG_DB_ID,
+  },
+  visualization_settings: {
+    fields: {
+      [TEST_PARAMETER.id]: {
+        id: TEST_PARAMETER.id,
+        required: true,
+        fieldType: "number",
+        inputType: "number",
+      },
+    },
+  },
+};
+
+describe("scenarios > models > actions", () => {
+  beforeEach(() => {
+    restore("postgres-12");
+    cy.signInAsAdmin();
+
+    enableActionsForDB();
+    enableActionsForDB(PG_DB_ID);
+
+    cy.createQuestion(SAMPLE_ORDERS_MODEL, {
+      wrapId: true,
+      idAlias: "modelId",
+    });
+
+    cy.intercept("/api/card/*").as("getModel");
+  });
+
+  it("should allow to view, create and edit model actions", () => {
+    cy.get("@modelId").then(id => {
+      cy.visit(`/model/${id}/detail`);
+      cy.wait("@getModel");
+    });
+
+    cy.findByText("Actions").click();
+
+    cy.findByRole("button", { name: /Create basic actions/i }).click();
+    cy.findByLabelText("Action list").within(() => {
+      cy.findByText("Create").should("be.visible");
+      cy.findByText("Update").should("be.visible");
+      cy.findByText("Delete").should("be.visible");
+    });
+
+    cy.findByRole("button", { name: "New action" }).click();
+    fillQuery("DELETE FROM orders WHERE id = {{ id }}");
+    fieldSettings()
+      .findByText(/Number/i)
+      .click();
+    cy.findByRole("button", { name: "Save" }).click();
+    modal().within(() => {
```

Comment:
Hmmm, I'm not sure if this is intended, but if `cy.get` returns 2 or more elements, `.within` should fail.

<img width="1304" alt="image" src="https://user-images.githubusercontent.com/1937582/215195762-7fd31d1f-fbe2-4ce2-a46d-88253f6641ba.png">

When saving a new action, we actually are opening 2 modals, but what `modal()` here is actually getting is 

https://github.com/metabase/metabase/blob/644011426b8bc0fbf3c6092ba507ed249b0072be/frontend/test/__support__/e2e/helpers/e2e-ui-elements-helpers.js#L9-L11

And the action creator modal doesn't seem to have `.ModalContent` class. That's why I think `modal()` only gets the save new action modal here, not the action creator modal.

All I want to say is it could be confusing why this is working since we are opening multiple modals.

Reply:
`modal` helper looks for CSS classes of `ModalContent` which isn't used by `ActionCreator` :/ So here it only finds the "New Action" modal. It'd probably be worth to make `modal` use `findByRole("dialog")` or something